pypeit/spectrographs/keck_lris.py

```diff
@@ -127,20 +130,40 @@ def init_meta(self):
         # Red only, but grabbing here
         meta['dispangle'] = dict(ext=0, card='GRANGLE', rtol=1e-2)
 
-        # Lamps
-        lamp_names = ['MERCURY', 'NEON', 'ARGON', 'CADMIUM', 'ZINC', 'KRYPTON', 'XENON',
-                      'FEARGON', 'DEUTERI', 'FLAMP1', 'FLAMP2', 'HALOGEN']
-        for kk,lamp_name in enumerate(lamp_names):
-            meta['lampstat{:02d}'.format(kk+1)] = dict(ext=0, card=lamp_name)
+        # Lamps -- Have varied in time..
+        for kk in range(12): # This needs to match the length of LAMPS below
+            meta['lampstat{:02d}'.format(kk+1)] = dict(card=None, compound=True)
         # Ingest
         self.meta = meta
 
     def compound_meta(self, headarr, meta_key):
         if meta_key == 'binning':
-#            return '1,1'
             binspatial, binspec = parse.parse_binning(headarr[0]['BINNING'])
             binning = parse.binning2string(binspec, binspatial)
             return binning
+        elif 'lampstat' in meta_key:
+            idx = int(meta_key[-2:])
+            curr_date = time.Time(headarr[0]['MJD-OBS'], format='mjd')
+            # Modern
+            t_newlamp = time.Time("2010-06-01", format='isot')  # LAMPS changed in Header
```

Comment:
Advise March 16 2010 for the lamp change given the news note indicating that's when the lamp position was moved.

Reply:
done.  And we'll modify it later as needed.